src/api/events/window.ts

```diff
@@ -32,6 +32,16 @@ export interface WindowReloadedEvent<Topic, Type> extends WindowEvent<Topic, Typ
     url: string;
 }
 
+export interface WindowOptionsChanged<Topic, Type> extends WindowEvent<Topic, Type> {
+    diff: WindowOptionDiffObject[];
+}
+
+export interface WindowOptionDiffObject {
```

Comment:
We could make WindowOptionDiffObject take a generic type which extends `keyof WindowOptions` and then make oldVal and newVal the type of that key on WindowOptions